File: QTD.SPA/src/app/_DtoModels/ClientUserSettingsDashboard/CustomizeDashboardUpdateOptions.ts
export class CustomizeDashboardUpdateOptions{

    updates: Array<CustomDashboardSettingOption>

    UpdateSettingOption = (setting: CustomDashboardSettingOption) => {
        if (!this.updates) this.updates = [];
        let updateSetting = this.updates.filter(r => r.settings == setting.settings)[0];
        if (updateSetting) {
            updateSetting.enable = setting.enable;
            updateSetting.disable = setting.disable;
          } else {
            this.updates.push(setting);
          }
    }
}

export class CustomDashboardSettingOption {
    settings:string;
    enable: boolean;
    disable: boolean

    constructor(setting: string, enable: boolean, disable: boolean){
        this.settings = setting;
        this.enable = enable;
        this.disable = disable;
    }
}

File: QTD2.Infrastructure/Model/ClientUserSettings/CustomizeDashboardUpdateOptions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.ClientUserSettings
{
    public class CustomizeDashboardUpdateOptions
    {
        public List<CustomDashboardSettingOption> Updates { get; set; }
    }

    public class CustomDashboardSettingOption
    {
        public string Settings { get; set; }
        public bool Enable { get; set; }
        public bool Disable { get; set; }

    }
}
